### PR TITLE
Feat(MicroSplit): Noise Model Trainer

### LINKED ISSUE #850: MicroSplit API: NoiseModelTrainer should produce a MultiChannelNMConfig directly
## Problem

NoiseModelTrainer.get_multichannel_model() returns a MultiChannelNoiseModel (runtime nn.Module), but the configuration pipeline needs a MultiChannelNMConfig (Pydantic model). There is no method to go from trained models back to a config object without saving/loading from disk.

## Potential solution

Add NoiseModelTrainer.get_config() -> MultiChannelNMConfig that extracts weights from trained GaussianMixtureNoiseModel instances into GaussianMixtureNMConfig objects. This should close the loop: train -> get config -> pass to factory.

### PR DESCRIPTION
**Pull request by CatEek**
## Disclaimer

- [ ] I am an AI agent.
- [x] I have used AI and I thoroughly reviewed every line.
- [ ] I have not used AI extensively.

## Description

> [!NOTE]  
> **tldr**: Noise Model training is now done in a separate class

### Background - why do we need this PR?

Noise-model training and usage had several friction points:
- Brittle separate code existed in the notebook
- Little to no validation existed

These issues made training brittle and delayed feedback on invalid setups.

### Overview - what changed?

This PR makes noise-model handling more explicit and robust by:
- Removing N2V internals from `NoiseModelTrainer` and keeping dedicated pair-based training.
- Extending `train_from_pairs(...)` to accept per-input axes and normalize both inputs before shape validation.
- Adding config reconstruction from saved `.npz` noise-model files.
- Enforcing noise-model presence at fit start when denoiSplit requires it, with the existing step-time guard retained as a backstop.

### Implementation - how did you implement the changes?

- `NoiseModelTrainer`:
  - Removed internal N2V-dependent training flow.
  - Kept `train_from_pairs(signal, observation, ...)` as the training entrypoint.
  - Added optional `signal_axes` and `observation_axes` arguments and normalized inputs via `reshape_array(...)` into canonical `SC(Z)YX` before compatibility checks.
- Added `NoiseModelTrainer.config_from_paths(paths)`:
  - Builds `MultiChannelNMConfig` directly from saved `.npz` files.
  - Propagates `channel_indices` when available.
  - Raises a clear error for empty path lists.
- `VAEModule`:
  - Added `_validate_noise_model_required()` and invoked it from `on_fit_start()`.
  - Ensures early, actionable failure when `denoisplit_weight > 0` and no noise model is attached.

## Changes Made

- `config_from_paths(paths)` utility in `NoiseModelTrainer` for rebuilding `MultiChannelNMConfig` from persisted noise-model files.

### Modified features or files

- `NoiseModelTrainer.train_from_pairs(...)`: added `signal_axes`/`observation_axes` support with normalization-before-validation.
- `VAEModule.on_fit_start()`: added early validation for required noise model when denoiSplit is active.
- `tests/noise_model/test_noise_model_trainer.py`: added/updated tests for axes normalization, mismatch behavior, config reconstruction, fit-start guard, runtime attachment, and musplit-only no-noise-model-required behavior.

## How has this been tested?

- Ran focused trainer tests around `train_from_pairs` behavior.
- Ran focused `set_noise_model` / `on_fit_start` guard tests.
- Ran a broader fast subset of `tests/noise_model/test_noise_model_trainer.py`.
- Checked lint diagnostics for touched files and resolved issues.

## Related Issues

- Resolves #850 

## Additional Notes and Examples

Will add a code snippet
---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [x] New tests have been added (for bug fixes/features)
- [x] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

### CODE REVIEW

File: src/careamics/config/noise_model/noise_model_config.py
Comment (jdeschamps) [inline src/careamics/config/noise_model/noise_model_config.py:1]: Unused Tensor annotation?

File: src/careamics/lightning/microsplit_data_module.py [hunk 1]
```diff
@@ -404,6 +404,7 @@ def create_microsplit_train_datamodule(
         "batch_size": batch_size,
         "train_dataloader_params": train_dataloader_params,
         "val_dataloader_params": val_dataloader_params,
+        "input_is_sum": True,  # TODO remove, hardcoded for debug
```
Comment (jdeschamps) [inline src/careamics/lightning/microsplit_data_module.py]: Remove

File: src/careamics/lightning/lightning_module.py [hunk 2]
```diff
@@ -467,15 +473,89 @@ def forward(self, x: torch.Tensor) -> tuple[torch.Tensor, dict[str, Any]]:
         """
         return self.model(x)  # TODO Different model can have more than one output
 
-    def set_noise_model(self, noise_model: MultiChannelNoiseModel) -> None:
-        """Set the noise model.
+    def set_noise_model(
+        self,
+        noise_model: (
+            MultiChannelNoiseModel | MultiChannelNMConfig | list[str] | list[Path]
```
Comment (jdeschamps) [inline src/careamics/lightning/lightning_module.py]:
```suggestion
            MultiChannelNoiseModel | list[str] | list[Path]
```

File: src/careamics/lightning/lightning_module.py [hunk 3]
```diff
@@ -467,15 +473,89 @@ def forward(self, x: torch.Tensor) -> tuple[torch.Tensor, dict[str, Any]]:
...
+
+        Accepts a ready-to-use ``MultiChannelNoiseModel``, a
+        ``MultiChannelNMConfig`` Pydantic model, or an ordered list of paths
+        to ``.npz`` files (one per output channel, in channel order).
 
         Parameters
         ----------
-        noise_model : MultiChannelNoiseModel
-            The noise model to use.
+        noise_model : MultiChannelNoiseModel | MultiChannelNMConfig | list[str | Path]
```
Comment (jdeschamps) [inline src/careamics/lightning/lightning_module.py]:
```suggestion
        noise_model : MultiChannelNoiseModel | list[str | Path]
```

File: src/careamics/lightning/lightning_module.py [hunk 4]
```diff
@@ -467,15 +473,89 @@ def forward(self, x: torch.Tensor) -> tuple[torch.Tensor, dict[str, Any]]:
...
+                "not require a noise model (denoisplit_weight <= 0). "
+                "Only denoiSplit configurations with denoisplit_weight > 0 use "
+                "a noise model."
+            )
+
+        # --- normalise input to MultiChannelNoiseModel -----------------
+        if isinstance(noise_model, MultiChannelNoiseModel):
+            resolved: MultiChannelNoiseModel = noise_model
+        elif isinstance(noise_model, MultiChannelNMConfig):
+            resolved = multichannel_noise_model_factory(noise_model)
```
Comment (jdeschamps) [inline src/careamics/lightning/lightning_module.py]:
```suggestion
```

File: src/careamics/lightning/modules/vae_lightning_module.py
Comment (jdeschamps) [inline src/careamics/lightning/modules/vae_lightning_module.py:1]: In `__init__`, `self.noise_nodel` should probably not be created.

**Review by jdeschamps — CHANGES_REQUESTED** [review]

File: src/careamics/lightning/modules/vae_lightning_module.py
Comment (CatEek) [reply src/careamics/lightning/modules/vae_lightning_module.py:1]: Created an issue. I'd not touch it here because it creates some other changes into the LM that are beyond the scope of this PR

**Review by jdeschamps — APPROVED** [review]
